### PR TITLE
Add support for content:// URI in CompressedExplorerFragment

### LINKED ISSUE #2650: StringIndexOutOfBoundsException on opening compressed file
## Exception
* __App Name:__ Amaze File Manager
* __Package:__ com.amaze.filemanager
* __Version:__ 3.6.1
* __User Action:__ UI Error
* __Request:__ Application crash
* __OS:__ Linux Android 11 - 30
* __Device:__ 1915
* __Model:__ vivo 1915
* __Product:__ 1915
<details><summary><b>Crash log </b></summary><p>

```
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
at java.lang.String.substring(String.java:2064)
at
com.amaze.filemanager.ui.fragments.CompressedExplorerFragment.onActivityCreated(CompressedExplorerFragment.java:221)
at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:2983)
at
androidx.fragment.app.FragmentStateManager.activityCreated(FragmentStateManager.java:577)
at
androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:285)
at
androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2177)
at
androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2088)
at
androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1990)
at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:239)
at android.app.ActivityThread.main(ActivityThread.java:8205)
at java.lang.reflect.Method.invoke(Native Method)
at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:626)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1015)

```
</details>
<hr>

### PR DESCRIPTION
**Pull request by TranceLove**
## Description
As a band-it solution to fix #2650.

When a content:// URI is given, it will query the filename via the ContentResolver, then pass the URI's InputStream to a cache file, which shall be deleted on fragment's destruction or process exit.

Will Toast user if given URI cannot be opened.

There is a downside to this approach. As we will be capturing the whole InputStream for the specified URI, it will become a problem if the content URI's file size is very big.

#### Issue tracker   
Fixes #2650

#### Automatic tests
- [ ] Added test cases
  
#### Manual tests
- [x] Done  
  
Device: Pixel 2 emulator
OS: Android 11

Test scenario: open a just downloaded file from Chrome or other web browser, which would trigger providing a content:// URI or downloaded file to CompressedExplorerFragment. It should not crash.

Alternately, try open a compressed file for viewing from Amaze itself. It should not crash either.

#### Build tasks success  
Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

**Comment by TranceLove** [comment]
Sorry but code is pending Kotlintizing and integrate `ViewModel` into the fragment. If code is on a hurry I can fix the spotless problem first though.

**Comment by VishalNehra** [comment]
Getting following error if i rotate the screen and rotate back to original orientation. If not related to this PR everything else seems fine 

## Issue explanation (write below this line)



## Exception
* __App Name:__ Amaze File Manager
* __Package:__ com.amaze.filemanager.debug
* __Version:__ 3.6.1
* __User Action:__ UI Error
* __Request:__ Application crash
* __OS:__ Linux Android 11 - 30
* __Device:__ generic_x86_arm
* __Model:__ sdk_gphone_x86
* __Product:__ sdk_gphone_x86
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: Unable to destroy activity {com.amaze.filemanager.debug/com.amaze.filemanager.ui.activities.MainActivity}: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5111)
at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5140)
at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5432)
at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5362)
at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:223)
at android.app.ActivityThread.main(ActivityThread.java:7656)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
at java.util.ArrayList.get(ArrayList.java:437)
at com.amaze.filemanager.ui.fragments.CompressedExplorerFragment.onDestroyView(CompressedExplorerFragment.kt:458)
at androidx.fragment.app.Fragment.performDestroyView(Fragment.java:3187)
at androidx.fragment.app.FragmentStateManager.destroyFragmentView(FragmentStateManager.java:739)
at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:346)
at androidx.fragment.app.SpecialEffectsController$FragmentStateManagerOperation.complete(SpecialEffectsController.java:742)
at androidx.fragment.app.SpecialEffectsController$Operation.cancel(SpecialEffectsController.java:594)
at androidx.fragment.app.SpecialEffectsController.forceCompleteAllOperations(SpecialEffectsController.java:329)
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3116)
at androidx.fragment.app.FragmentManager.dispatchDestroy(FragmentManager.java:3091)
at androidx.fragment.app.FragmentController.dispatchDestroy(FragmentController.java:334)
at androidx.fragment.app.FragmentActivity.onDestroy(FragmentActivity.java:322)
at androidx.appcompat.app.AppCompatActivity.onDestroy(AppCompatActivity.java:242)
at com.amaze.filemanager.ui.activities.MainActivity.onDestroy(MainActivity.java:1358)
at android.app.Activity.performDestroy(Activity.java:8245)
at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1344)
at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5096)
... 13 more

```
</details>
<hr>

**Review by VishalNehra — APPROVED** [review]

**Review by VishalNehra — APPROVED** [review]

File: app/src/main/java/com/amaze/filemanager/ui/fragments/CompressedExplorerFragment.kt [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+import com.github.junrar.exception.UnsupportedRarV5Exception
+import com.google.android.material.appbar.AppBarLayout
+import com.google.android.material.appbar.AppBarLayout.OnOffsetChangedListener
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.*
+
+@Suppress("TooManyFunctions")
+class CompressedExplorerFragment : Fragment(), BottomBarButtonPath {
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/ui/fragments/CompressedExplorerFragment.kt:76]: There are quite a few `!!` used here, this should be fixed.

Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/ui/fragments/CompressedExplorerFragment.kt:76]: > There are quite a few `!!` used here, this should be fixed.

See #2796

**Pull request merged**